crates/edr_eth/src/remote/client.rs

```diff
@@ -220,27 +220,39 @@ impl RpcClient {
         })
     }
 
-    fn parse_response_str<T: DeserializeOwned>(response: &str) -> Result<T, RpcClientError> {
-        serde_json::from_str(response).map_err(|error| RpcClientError::InvalidResponse {
-            response: response.to_string(),
-            expected_type: std::any::type_name::<T>(),
+    fn parse_response_str<T: DeserializeOwned>(
+        response: String,
+    ) -> Result<jsonrpc::Response<T>, RpcClientError> {
+        serde_json::from_str(&response).map_err(|error| RpcClientError::InvalidResponse {
+            response,
+            expected_type: std::any::type_name::<jsonrpc::Response<T>>(),
             error,
         })
     }
 
-    fn extract_result<T: DeserializeOwned>(
+    async fn retry_on_sporadic_failure<T: DeserializeOwned>(
+        &self,
+        error: jsonrpc::Error,
         request: SerializedRequest,
-        response: String,
     ) -> Result<T, RpcClientError> {
-        let response: jsonrpc::Response<T> = Self::parse_response_str(&response)?;
+        fn is_missing_trie_node_error(error: &jsonrpc::Error) -> bool {
+            error.code == -32000 && error.message.contains("missing trie node")
+        }
 
-        response
-            .data
-            .into_result()
-            .map_err(|error| RpcClientError::JsonRpcError {
-                error,
-                request: request.to_json_string(),
-            })
+        let result = if is_missing_trie_node_error(&error) {
+            self.send_request_body(&request)
```

Comment:
Should we wait some milliseconds here before re-sending? I'm fine with not doing it, just wondering out loud.

Reply:
I don't think it matters in this case. At the middleware layer we're already retrying with a delay to avoid too many requests, so it would be duplicating part of that logic.

Reply:
https://github.com/NomicFoundation/edr/pull/410/commits/0519bc68a5dbed5ac60ba299d30712f5cb63e233